aperturedb/ImageLoader.py

```diff
@@ -86,11 +118,50 @@ def load_image(self, filename):
 
         return False, None
 
+    def load_url(self, url):
+
+        retries = 0
+        while True:
+            imgdata = requests.get(url)
+            if imgdata.ok:
+                if self.check_image:
+                    filename = "/tmp/" + url.split("/")[-1]
+                    fd = open(filename, "wb")
+                    fd.write(imgdata.content)
+                    fd.close()
+                    self.check_valid_image(filename)
+                    os.remove(filename)  # no need to keep it around
+                return imgdata.ok, imgdata.content
+            else:
+                if retries >= self.n_download_retries:
+                    break
+                print("WARNING: Retrying object:", url)
+                retries += 1
+                time.sleep(2)
+
+        return False, None
+
+    def load_s3_url(self, s3_url):
+        s3 = boto3.client('s3')
+
+        try:
+            filename = "/tmp/" + s3_url.split("/")[-1]
+            bucket_name = s3_url.split("/")[2]
+            object_name = s3_url.split("s3://" + bucket_name + "/")[-1]
+            s3.download_file(bucket_name, object_name, filename)
```

Comment:
it seems you can avoid downloading to a file here as well:
https://stackoverflow.com/questions/37087203/retrieve-s3-file-as-object-instead-of-downloading-to-absolute-system-path